students/shawn_degraw/session07/html_render.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+class Element(object):
+    """ Default Element class for rendering html """
+    tag_name = "html"
+    indent = "   "
+    cur_ind = ""
+
+    def __init__(self, content=None, **kwargs):
+        self.html_content = ([content] if content else [])
+        self.header_arg = (kwargs if kwargs else {})
+
+    def append(self, new_content):
+        self.html_content.append(new_content)
+
+    def render(self, out_file, cur_ind=""):
+        if self.header_arg:
+            self.writetag(out_file, cur_ind + "<{}")
+            for k, v in self.header_arg.items():
+                out_file.write(" {}=\"{}\"".format(k, v))
+            out_file.write(">\n")
+        else:
+            self.writetag(out_file, cur_ind + "<{}>\n")
```

Comment:
Looks like the `if` condition and this `else` block are redundant, because if `self.header_arg` is empty, then the `for` loop will not iterate at all, thus giving the same output for the `else` case, even with the code in the `if` block.

Reply:
Interesting.  I assumed the for loop would do something even when the header_arg was empty.  Took out the if/else and everything still worked fine!